tests/test_config_parsing/test_ert_config_parsing.py

```diff
@@ -1021,3 +1026,35 @@ def test_that_multiple_errors_are_shown_for_forward_model():
 
     assert expected_nice_message == cli_message
     assert error_messages_list == expected_nice_messages_list
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_redefines_are_applied_correctly():
+    test_config_file_name = "test.ert"
+    test_config_contents = dedent(
+        """
+        NUM_REALIZATIONS  1
+        DEFINE <A> 2
+        DEFINE B <A>
```

Comment:
Would be good to test that intermediate usages are substituted. So try a multioccurrance `KW <A>` and check that it first gets argument `2` and then later `3`.